src/sync/mod.rs

```diff
@@ -139,6 +148,114 @@ pub enum BeginSyncError {
     WrongChecksum(String),
 }
 
+pub async fn maybe_end_sync(
+    store: &dag::Store,
+    maybe_end_sync_req: &MaybeEndSyncRequest,
+) -> Result<MaybeEndSyncResponse, MaybeEndSyncError> {
+    use MaybeEndSyncError::*;
+
+    // Ensure sync head is what the caller thinks it is.
+    let mut dag_write = store.write().await.map_err(WriteError)?;
+    let dag_read = dag_write.read();
+    let sync_head_hash = dag_read
+        .get_head(SYNC_HEAD_NAME)
+        .await
+        .map_err(ReadError)?
```

Comment:
`GetSyncHeadError`?

Reply:
done in https://github.com/rocicorp/repc/pull/193